network/src/connectivity_manager/mod.rs

```diff
@@ -70,9 +70,8 @@ pub struct ConnectivityManager<TTicker, TSubstream, TBackoff> {
 pub enum ConnectivityRequest {
     /// Request to update known addresses of peer with id `PeerId` to given list.
     UpdateAddresses(PeerId, Vec<Multiaddr>),
-    /// Update set of nodes eligible to join the network.
-    UpdateEligibleNodes(HashMap<PeerId, NetworkPublicKeys>),
```

Comment:
But we still want `self.eligible` to be updated if the set of trusted peers changes, right? Since trusted peers is shared b/w discovery and connectivity manager anyway, I thought there's no harm in sharing with NetworkProvider as well.
Are you suggesting that we stick with the message passing syntax? I'm honestly fine either way - the reason to go for `Arc` over message passing was that now we'd need to pass an `Option<channel::Sender<ConnecitivityRequest>>` to network provider since connectivity manager is not always created.

Reply:
yeah I'd prefer message passing

Reply:
I've removed this commit for now -- agree with your overall idea though, so I think we should also get ride of the shared state b/w Discovery and ConnectivityManager. Ideally the message flow should be `NetworkProvider` -> `Discovery` -> `ConnectivityManager`. Will file an issue to track this.

Reply:
https://github.com/libra/libra/issues/775